opentelemetry-prometheus/src/utils.rs

```diff
@@ -0,0 +1,201 @@
+use opentelemetry_api::metrics::Unit;
+use std::borrow::Cow;
+
+const NON_APPLICABLE_ON_PER_UNIT: [&str; 8] = ["1", "d", "h", "min", "s", "ms", "us", "ns"];
+
+pub(crate) fn get_unit_suffixes(unit: &Unit) -> Option<Cow<'static, str>> {
+    // no unit return early
+    if unit.as_str().is_empty() {
+        return None;
+    }
+
+    // direct match with known units
+    if let Some(matched) = get_prom_units(unit.as_str()) {
+        return Some(Cow::Borrowed(matched));
+    }
+
+    // converting foo/bar to foo_per_bar
+    // split the string by the first '/'
+    // if the first part is empty, we just return the second part if it's a match with known per unit
+    // e.g
+    // "test/y" => "per_year"
+    // "km/s" => "kilometers_per_second"
+    if unit.as_str().contains('/') {
+        let components = unit.as_str().splitn(2, '/').collect::<Vec<&str>>();
```

Comment:
nit: you could use https://doc.rust-lang.org/std/string/struct.String.html#method.split_once here instead of allocating a new vec

Reply:
Good point :+1: . Updated